sdk/core/azure-core/test/ut/transport_adapter.cpp

```diff
@@ -26,6 +26,88 @@ namespace Azure { namespace Core { namespace Test {
   Azure::Core::Http::HttpPipeline TransportAdapter::pipeline(policies);
   Azure::Core::Context TransportAdapter::context = Azure::Core::GetApplicationContext();
 
+  // multiThread test requires `ConnectionsOnPool` hook which is only available when building
+  // TESTING_BUILD. This test cases are only built when that case is true.`
+  TEST_F(TransportAdapter, getMultiThread)
+  {
+    std::string host("http://httpbin.org/get");
+    Azure::Core::Http::CurlConnectionPool::ClearIndex();
+
+    auto threadRoutine = [host]() {
+      auto request = Azure::Core::Http::Request(Azure::Core::Http::HttpMethod::Get, host);
+      auto response = pipeline.Send(context, request);
+      checkResponseCode(response->GetStatusCode());
+      auto expectedResponseBodySize = std::stoull(response->GetHeaders().at("content-length"));
+      CheckBodyFromBuffer(*response, expectedResponseBodySize);
+    };
+
+    std::thread t1(threadRoutine);
+    std::thread t2(threadRoutine);
+    t1.join();
+    t2.join();
+
+    // 2 connections must be available at this point
+    EXPECT_EQ(Http::CurlConnectionPool::ConnectionsOnPool("httpbin.org"), 2);
+
+    std::thread t3(threadRoutine);
+    std::thread t4(threadRoutine);
+    std::thread t5(threadRoutine);
+    t3.join();
+    t4.join();
+    t5.join();
+    // wait a few ms for connections to go back to pool.
+    std::this_thread::sleep_for(std::chrono::milliseconds(1000));
+    // Two connections re-used plus one connection created
+    EXPECT_EQ(Http::CurlConnectionPool::ConnectionsOnPool("httpbin.org"), 3);
+  }
+
+#ifdef RUN_LONG_UNIT_TESTS
+  TEST_F(TransportAdapter, ConnectionPoolCleaner)
+  {
+    std::string host("http://httpbin.org/get");
+
+    auto threadRoutine = [host]() {
+      auto request = Azure::Core::Http::Request(Azure::Core::Http::HttpMethod::Get, host);
+      auto response = pipeline.Send(context, request);
+      checkResponseCode(response->GetStatusCode());
+      auto expectedResponseBodySize = std::stoull(response->GetHeaders().at("content-length"));
+      CheckBodyFromBuffer(*response, expectedResponseBodySize);
+    };
+
+    // 3 connections from previous test. Make sure cleaner remove them
+    EXPECT_EQ(Http::CurlConnectionPool::ConnectionsOnPool("httpbin.org"), 3);
+
+    std::cout
+        << "Running Connection Pool Cleaner Test. This test takes more than 3 minutes to complete."
+        << std::endl
+        << "Add compiler option -DRUN_LONG_UNIT_TESTS=OFF when building if you want to skip this "
+           "test."
+        << std::endl;
+
+    // Wait for 100 secs to make sure any previous connection is removed by the cleaner
+    std::this_thread::sleep_for(std::chrono::milliseconds(1000 * 180));
```

Comment:
Why did we increase this? And update the  comment.
```suggestion
    // Wait for 180 secs to make sure any previous connection is removed by the cleaner
    std::this_thread::sleep_for(std::chrono::milliseconds(1000 * 180));
```

Reply:
This test is ran after the multi thread test. It validates and expects 3 connections to be on the pool when test starts.
Then we need to wait some time for the cleaner to make its job.  Cleaner runs every 90 sec, so the initial value of 100 sec was hoping to be enough to see the cleaner results.
However, connections will expire when they have been waiting for 90 sec, so it could happen that on the first clean routine, connections are not expired yet. That's why I increased to make sure cleaner has enough time to find connections expired and remove them